source/persistence/MessageController.h

```diff
@@ -19,7 +19,7 @@ class MessageController : public QSqlTableModel
     virtual QHash<int, QByteArray> roleNames() const;
 
     void setFilterOnJid(QString const &jidFiler);
-    bool addMessage(const QString &id, QString const &jid, const QString &resource, QString const &message, const QString &type, unsigned int direction, qint64 timestamp = 0);
+    bool addMessage(const QString &id, QString const &jid, const QString &resource, QString const &message, const QString &type, bool isGroupMessage, unsigned int direction, qint64 timestamp = 0);
```

Comment:
try to get rid of the additional attribute in the database. See the comment at main.qml